proto/attestation/reference_value.proto

```diff
@@ -25,81 +25,141 @@ option go_package = "proto/oak/attestation/v1";
-  bytes rekor_public_key = 2;
-}
-
 // Verifies that the field contains at least one of the given digests.
-// No checks are performed if this is empty.
+// No checks are performed if this is empty. A match in at least one
+// digest is considered a success.
 message Digests {
   repeated RawDigest digests = 1;
 }
 
 // Reference values of the kernel layer, as measured by stage0.
-message KernelLayerReferenceValues {
+message KernelReferenceValues {
   // Verifies the kernel binary based on endorsement.
-  optional LogEntryVerification log_entry = 1;
+  BinaryReferenceValue kernel_image = 1;
 
-  // Explicitly verifies digests in addition to log entry verification.
-  // Note that endorsement-based digest verification is implicitly contained
-  // in the log entry verification.
-  optional Digests code = 2;
+  // Compare command line as string until further notice.
+  StringReferenceValue kernel_cmd_line = 2;
 
-  // Verifies that the command line is equivalent to a specified one (e.g. up
-  // to certain permutations).
-  repeated string kernel_cmd_line = 3;
+  BinaryReferenceValue kernel_setup_data = 3;
 
-  optional Digests kernel_setup_data = 4;
+  BinaryReferenceValue init_ram_fs = 4;
 
-  optional Digests init_ram_fs = 5;
+  BinaryReferenceValue memory_map = 5;
 
-  repeated Digests memory_map = 6;
+  BinaryReferenceValue acpi = 6;
+}
 
-  optional Digests acpi = 7;
+message InitReferenceValues {
+  // Verifies the binary based on endorsement.
+  BinaryReferenceValue binary = 1;
+
+  // Configuration measurements.
+  BinaryReferenceValue configuration = 2;
 }
 
-message CustomLayerReferenceValues {
-  // Verifies the layer binary based on endorsement.
-  optional LogEntryVerification log_entry = 1;
+// Represents an application running under Oak Restricted Kernel.
+message ApplicationReferenceValues {
+  // Verifies the binary based on endorsement.
+  BinaryReferenceValue binary = 1;
 
-  // Explicitly verifies digests in addition to log entry verification.
-  // Note that endorsement-based digest verification is implicitly contained
-  // in the log entry verification.
-  optional Digests code = 2;
+  // Configuration measurements.
+  BinaryReferenceValue configuration = 2;
+}
 
-  optional Digests configuration = 3;
+message ContainerReferenceValues {
+  // Verifies the binary based on endorsement.
+  BinaryReferenceValue binary = 1;
+
+  // Configuration measurements.
+  BinaryReferenceValue configuration = 2;
+}
+
+message OakRestrictedKernelReferenceValues {
+  RootReferenceValues root_layer = 1;
+  KernelReferenceValues kernel_layer = 2;
+  ApplicationReferenceValues application_layer = 3;
+}
+
+message OakContainersReferenceValues {
+  RootReferenceValues root_layer = 1;
+  KernelReferenceValues kernel_layer = 2;
+  InitReferenceValues init_layer = 3;
+  ContainerReferenceValues container_layer = 4;
+}
+
+message CBReferenceValues {
+  repeated CBReferenceValue layers = 1;
+}
+
+message CBReferenceValue {
```

Comment:
I think right now we have `CBReferenceValue` and `CBReferenceValues`. To avoid confusion we could rename `CBReferenceValue` to `CBLayerReferenceValue`.

And this could go for all other reference values as well, when we refer to the layer we add `Layer` to the name. So it's easier to understand what a proto message represents.

Reply:
Added `Layer` to names.